cmd/block-generator/generator/generate.go

```diff
@@ -368,60 +369,47 @@ func (g *generator) initializeAccounting() {
 }
 
 // convert wraps the transaction as needed for including in the block.
-func convert(txn sdk_types.Transaction, ad types.ApplyData) types.SignedTxnInBlock {
-	stxn := sdk_types.SignedTxn{
-		Sig:      sdk_types.Signature{},
-		Msig:     sdk_types.MultisigSig{},
-		Lsig:     sdk_types.LogicSig{},
+func convert(txn transactions.Transaction, ad transactions.ApplyData) transactions.SignedTxnInBlock {
+	stxn := transactions.SignedTxn{
+		Sig:      crypto.Signature{},
+		Msig:     crypto.MultisigSig{},
+		Lsig:     transactions.LogicSig{},
 		Txn:      txn,
-		AuthAddr: sdk_types.Address{},
+		AuthAddr: basics.Address{},
 	}
 
 	// TODO: Would it be useful to generate a random signature?
 	stxn.Sig[32] = 50
 
-	withAd := types.SignedTxnWithAD{
+	withAd := transactions.SignedTxnWithAD{
 		SignedTxn: stxn,
 		// TODO: Add close-amount to apply data
 		ApplyData: ad,
 	}
 
-	stxnib := types.SignedTxnInBlock{
+	stxnib := transactions.SignedTxnInBlock{
 		SignedTxnWithAD: withAd,
 		HasGenesisID:    true,
 		HasGenesisHash:  true,
 	}
 	return stxnib
 }
 
-func (g *generator) getSuggestedParams(round uint64) sdk_types.SuggestedParams {
```

Comment:
I prefer the pattern where the suggested params are a function of the generator and everything else is a pure function. To address the missing first/last valid maybe this function could be replaced with something to generate the header and that can be passed around instead?